android/services/mongodb-remote-perftests/src/androidTest/java/com/mongodb/stitch/android/services/mongodb/performance/SyncMixedPerformanceTestDefinitions.kt

```diff
@@ -0,0 +1,312 @@
+import com.mongodb.stitch.android.services.mongodb.performance.SyncPerformanceTestUtils.Companion.doSyncPass
+import com.mongodb.stitch.android.services.mongodb.performance.SyncPerformanceTestUtils.Companion.performLocalUpdate
+import com.mongodb.stitch.core.services.mongodb.remote.ExceptionListener
+import com.mongodb.stitch.core.services.mongodb.remote.sync.DefaultSyncConflictResolvers
+import org.bson.BsonDouble
+import org.bson.BsonValue
+import org.bson.Document
+import org.bson.types.ObjectId
+
+class SyncMixedPerformanceTestDefinitions {
+    companion object {
+
+        /*
+         * Before: Perform remote insert of numDoc / 2 documents
+         *         Perform a local insert of numDoc / 2 documents
+         *         Ensure there are numConflict conflicts
+         * Test: Configure sync to sync on the inserted docs and perform a sync pass
+         * After: Ensure that the initial sync worked as expected
+         */
+        fun testInitialSync(testHarness: SyncPerformanceIntTestsHarness, runId: ObjectId) {
+            SyncPerformanceTestUtils.getConflictPercentages().forEach { conflictPercentage ->
+                performInitialSync(testHarness, runId, conflictPercentage)
+            }
+        }
+
+        fun performInitialSync(
+            testHarness: SyncPerformanceIntTestsHarness,
+            runId: ObjectId,
+            conflictPercentage: Double
+        ) {
+            val testName = "Mixed_InitialSync"
+
+            // Local variable for list of documents captured by the test definition closures below.
+            // This should change for each iteration of the test.
+            var documentIds: Set<BsonValue>? = null
+
+            testHarness.runPerformanceTestWithParams(
+                testName, runId,
+                beforeEach = { ctx, numDocs, docSize ->
+                    val docIDs = mutableSetOf<BsonValue>()
+                    val numEach = numDocs / 2
+
+                    // Generate the documents that are to be synced via R2L and remotely insert them
+                    val remoteDocs = SyncPerformanceTestUtils.generateDocuments(docSize, numEach)
+                    val remoteInsertResult = Tasks.await(ctx.testColl.insertMany(remoteDocs))
+                    SyncPerformanceTestUtils.assertIntsAreEqualOrThrow(
+                        remoteInsertResult.insertedIds.size,
+                        numEach)
+                    docIDs.addAll(remoteInsertResult.insertedIds.map { it.value })
+
+                    // Use numEach * pctConflict of the remote documents in a local insert
+                    val numConflicts = (numEach * conflictPercentage).toInt()
+                    var localDocs = remoteDocs.subList(0, numConflicts) +
+                        SyncPerformanceTestUtils.generateDocuments(docSize, numEach - numConflicts)
+                    val localInsertResult = Tasks.await(ctx.testColl.sync().insertMany(localDocs))
```

Comment:
[minor] the L2R-only tests perform the local insert for initial sync as part of the test definition rather than the test setup. We should probably do the same here, but we can keep the remote insert in the setup.